Source/Utilities/EventBusMixin.lua

```diff
@@ -0,0 +1,41 @@
+AuctionatorEventBusMixin = {}
+
+function AuctionatorEventBusMixin:Init()
+  self.registeredFrames = {}
+  self.sources = {}
+end
+
+function AuctionatorEventBusMixin:Register(listener, eventNames)
+  if listener == nil or listener.frame.EventUpdate == nil then
+    error("Attempted to register an invalid frame! EventUpdate function must be defined.")
+    return
+  end
+
+  for _, eventName in ipairs(eventNames) do
+    if self.registeredFrames[eventName] == nil then
+      self.registeredFrames[eventName] = {}
+    end
+
+    table.insert(self.registeredFrames[eventName], frame)
+  end
+end
+
+function AuctionatorEventBusMixin:RegisterSource(source, name)
+  self.sources[tostring(source)] = name
+end
+
+function AuctionatorEventBusMixin:Fire(source, eventName, eventData)
```

Comment:
`eventData` should be `...`

Reply:
Yep, that was a leftover from the old bus